File: apk/app/src/main/java/app/momentra/ui/theme/AvatarColors.kt
package app.momentra.ui.theme

import androidx.compose.ui.graphics.Color

/** Avatar variant colours for user initials. */
object AvatarColors {
    val warmBg = Color(0xFFFAEEDA)
    val warmText = Color(0xFF633806)

    val deepBg = Color(0xFFEEEDFE)
    val deepText = Color(0xFF3C3489)

    val goBg = Color(0xFFE1F5EE)
    val goText = Color(0xFF085041)

    fun variantFor(index: Int): Pair<Color, Color> = when (index % 3) {
        0 -> warmBg to warmText
        1 -> deepBg to deepText
        else -> goBg to goText
    }
}

File: momentra/momentra/AvatarColors.swift
//
//  AvatarColors.swift
//  momentra
//
//  Created by santosh on 19/04/26.
//

import SwiftUI

/// Avatar variant colors for user initials
enum AvatarColors {
    // MARK: - Warm Variant
    static let warmBg = Color(hex: "FAEEDA")
    static let warmText = Color(hex: "633806")
    
    // MARK: - Deep Variant
    static let deepBg = Color(hex: "EEEDFE")
    static let deepText = Color(hex: "3C3489")
    
    // MARK: - Go Variant
    static let goBg = Color(hex: "E1F5EE")
    static let goText = Color(hex: "085041")
    
    /// Returns a color pair based on user index or hash
    static func variant(for index: Int) -> (bg: Color, text: Color) {
        switch index % 3 {
        case 0: return (warmBg, warmText)
        case 1: return (deepBg, deepText)
        default: return (goBg, goText)
        }
    }
}
